### DISCUSSION TITLE
提示词问题

### DISCUSSION THREAD

**Discussion by woshiyanghai** [hhyo/Archery#2859] [General]
现在数据库sql查询时，提示词并好用。是否有更好的提示词方式。

**Comment by LeoQuote** [comment]
系统配置里可以自定义提示词，可参考默认提示词进行修改，默认提示词模板可参考此处的代码： https://github.com/hhyo/Archery/blob/d97ccdceda15cff9d415192879d9eed31d25c771/common/utils/openai.py#L15

**Reply by woshiyanghai** [reply]
是需要配置openai？

**Reply by woshiyanghai** [reply]
哦，可能我的描述引起了误会，我的意思是，比如我再写一个sql当中，我想要在db名字，表名字，有提示下拉列表，同一个sql中的好几个表关联的提示都不很好用

**Reply by woshiyanghai** [reply]
https://github.com/actiontech/sqle.git  在sql查询时，db，表，字段类型等提示就准确很多，是否可以参考下。

**Reply by LeoQuote** [reply]
哦这方面我就不是特别熟，感谢你提出的问题，欢迎PR

**Reply by woshiyanghai** [reply]
目前存在的问题就是需要先选择库，然后才可以把db名称，表名，字段名字，加入提示字中，但是有的时候，有跨库查询就比较难，sql中虽然写了这个db名字,依然无法获取到表名提示，而且存在 比如我输入一个yh  ，结果  ayah也在提示词中，这就有难

**Reply by woshiyanghai** [reply]
这个提示词这里还需要前端大神修改下，比如我本来我输入的是yhtest.ab  结果居然能把很多都提示出来，比如，ayhtestb这种都会提示出来，这是不合理的。
